### PR TITLE
Create Controller for Project and Description.

### PR DESCRIPTION
**Pull request by chelsea-chenxi1999**
Create Controller for Project and Description.

### CODE REVIEW

File: backend/src/main/java/com/itlize/joolemarketplace/controller/ProjectController.java
Comment (tak-oshima) [inline backend/src/main/java/com/itlize/joolemarketplace/controller/ProjectController.java:1]: We also need endpoints for adding and removing products in projects. To have this, we need to add appropriate methods to `ProjectServiceImpl` as well

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            Description createdDescription = descriptionService.createDescription(description);
+            return new ResponseEntity<>(createdDescription, HttpStatus.CREATED);
+        }
+        catch (RuntimeException e) {
+            Map<String, String> responseBody = new HashMap<>();
+            responseBody.put("error", "Description already exists");
+            responseBody.put("details", e.getMessage());
+            return new ResponseEntity<>(responseBody, HttpStatus.CONFLICT);
+        }
+    }
```
Comment (tak-oshima) [inline backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: I'm curious how this endpoint works without the `productId`. Were you able to test the endpoint?

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    @GetMapping
+    public ResponseEntity<?> getDescriptionByProduct(@RequestParam(value = "product", required = false) Product product) {
```
Comment (tak-oshima) [inline backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: This endpoint is requesting the entire product object as the parameter, which probably won't work..? A better approach I think would be to request the `Integer productId` and get the actual object from `ProductService`

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            responseBody.put("details", String.format("Description with product \"%s\" could not be found", product));
+            return new ResponseEntity<>(responseBody, HttpStatus.NOT_FOUND);
+        }
+    }
+
+    /*@GetMapping
+    public ResponseEntity<?> getAllDescriptions() {
+        List<Description> descriptions = descriptionService.getAllDescriptions();
+        return new ResponseEntity<>(descriptions, HttpStatus.OK);
+    }**/
```
Comment (tak-oshima) [inline backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: I think you forgot to uncomment this

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            Description updatedDescription = descriptionService.updateDescription(description);
+            return new ResponseEntity<>(updatedDescription, HttpStatus.OK);
+        }
+        catch (RuntimeException e) {
+            Map<String, String> responseBody = new HashMap<>();
+            responseBody.put("error", "Description not found");
+            responseBody.put("details", e.getMessage());
+            return new ResponseEntity<>(responseBody, HttpStatus.NOT_FOUND);
+        }
+    }
```
Comment (tak-oshima) [inline backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: Same concern as `createDescription`. My bad if you already tested it works

File: backend/src/main/java/com/itlize/joolemarketplace/controller/ProjectController.java
Comment (chelsea-chenxi1999) [reply backend/src/main/java/com/itlize/joolemarketplace/controller/ProjectController.java:1]: Okay!

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java (hunk 1 above)
Comment (chelsea-chenxi1999) [reply backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: I'll try with `productId`, just like deleteDescription

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java (hunk 2 above)
Comment (chelsea-chenxi1999) [reply backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: Okk. Thanks for the tips!

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java (hunk 3 above)
Comment (chelsea-chenxi1999) [reply backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: Do we need this method...? I'm not sure. Are we going to use it?

File: backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java (hunk 4 above)
Comment (chelsea-chenxi1999) [reply backend/src/main/java/com/itlize/joolemarketplace/controller/DescriptionController.java]: I'll give a try with `productId`

**Pull request closed without merging**